osu.Game/Screens/Select/BeatmapClearScoresDialog.cs

```diff
@@ -39,6 +39,28 @@ public BeatmapClearScoresDialog(BeatmapInfo beatmap, Action onCompletion)
             };
         }
 
+        public BeatmapClearScoresDialog(ScoreInfo score, Action onCompletion)
+        {
+            string accuracy = string.Format(score?.Accuracy % 1 == 0 ? @"{0:P0}" : @"{0:P2}", score?.Accuracy);
+
+            BodyText = $@"{score?.Beatmap?.Metadata?.Artist} - {score?.Beatmap?.Metadata?.Title} {Environment.NewLine} {score?.User} - Rank: {score?.Rank} - Max Combo: {score?.MaxCombo} - {accuracy} - {score?.Date.Date.ToShortDateString()}";
```

Comment:
I'm pretty sure these null propagation operators are a bit too lenient - `score == null` means catastrophe and it'll most likely just fail somewhere down below like `ScoreManager`. I'd guard (either with an assert, or just plain exception) and drop the `?`s next to `score?`s.

Reply:
Ok. I'll fix this. 

Just to make sure we are on the same page. So we want to make sure score is not null and if it is null we want it to show an error in the dialog or just exist it.

> Tests would be nice, but I imagine they might be a little complicated to write for this scenario.

And for the testing, I had some issue setting up test environment for this.

Reply:
`BeatmapInfo` already has a `ToString`. Check other usages in the code and use what they do.

Reply:
I have updated to 
`BodyText = $@"{score} {Environment.NewLine} Rank: {score.Rank} - Max Combo: {score.MaxCombo} - {accuracy}";`

This will use the ToString for `ScoreInfo` which also uses the  ToString of `BeatmapInfo`. The rank and max combo is not included in the ToString, so I have to do that if we still want the information to be displayed.


This will display:
`{User} playing {Artist} - {Title} ({Author}) [{Version}]`
`{score.Rank} - Max Combo: {score.MaxCombo} - {accuracy}`